File: src/main/java/com/sfh/pokeRogueBot/model/browser/gamejson/GameJsonNode.java
package com.sfh.pokeRogueBot.model.browser.gamejson;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.Map;

@Data
public class GameJsonNode {

    @SerializedName("biome")
    private String biome;
    @SerializedName("money")
    private int money;
    @SerializedName("wave")
    private int wave;
    @SerializedName("gameMode")
    private GameMode gameMode;
    @SerializedName("gameData")
    private GameData gameData;
    private Float gameSpeed;
    private Map<String, Integer> pokeballCounts;
    @SerializedName("currentBattle")
    private CurrentBattle currentBattle;

}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/browser/gamejson/GameJsonNode.kt
package com.sfh.pokeRogueBot.model.browser.gamejson

import com.google.gson.annotations.SerializedName

data class GameJsonNode(
    @SerializedName("biome")
    val biome: String?,

    @SerializedName("money")
    val money: Int,

    @SerializedName("wave")
    val wave: Int,

    @SerializedName("gameMode")
    val gameMode: GameMode?,

    @SerializedName("gameData")
    val gameData: GameData?,

    val gameSpeed: Float?,
    val pokeballCounts: Map<String, Int>?,

    @SerializedName("currentBattle")
    val currentBattle: CurrentBattle?
)